contentcuration/contentcuration/static/js/edit_channel/uploader/utils.js

```diff
@@ -0,0 +1,402 @@
+    const idx = answers.findIndex(answer => answer.correct);
+    return idx === -1 ? null : idx;
+  }
+
+  return answers
+    .map((answer, idx) => {
+      return answer.correct ? idx : undefined;
+    })
+    .filter(idx => idx !== undefined);
+};
+
+/**
+ * Updates `correct` fields of answers based on index/indexes stored in `correctAnswersIndice`.
+ * @param {Array} answers An array of answer objects { answer: ..., correct: ..., ...}
+ * @param {Number|null|Array} correctAnswersIndices A correct answer index or an array of
+ * correct answers indexes.
+ * @returns {Array} An array of answer objects with updated `correct` fields.
+ */
+export const mapCorrectAnswers = (answers, correctAnswersIndices) => {
+  if (!answers || !answers.length) {
+    return null;
+  }
+
+  return answers.map((answer, idx) => {
+    const isAnswerCorrect =
+      correctAnswersIndices === idx ||
+      (Array.isArray(correctAnswersIndices) && correctAnswersIndices.includes(idx));
+
+    return {
+      ...answer,
+      correct: isAnswerCorrect,
+    };
+  });
+};
+
+/**
+ * Update answers to correspond to a question type:
+ * - multiple selection: No answers updates needed.
+ * - input question: Make all answers correct.
+ * - true/false: Remove answers in favour of new true/false values.
+ * - single selection: Keep first correct choice only if there is any.
+ *                     Otherwise mark first choice as correct.
+ * @param {String} newQuestionType single/multiple selection, true/false, input question
+ * @param {Array} answers An array of answer objects.
+ * @returns {Array} An array of updated answer objects.
+ */
+export const updateAnswersToQuestionType = (questionType, answers) => {
+  const NEW_TRUE_FALSE_ANSWERS = [
+    { answer: translator.translate('true'), correct: true, order: 1 },
+    { answer: translator.translate('false'), correct: false, order: 2 },
+  ];
+
+  if (!answers || !answers.length) {
+    if (questionType === AssessmentItemTypes.TRUE_FALSE) {
+      return NEW_TRUE_FALSE_ANSWERS;
+    } else {
+      return [];
+    }
+  }
+
+  let answersCopy = JSON.parse(JSON.stringify(answers));
+
+  switch (questionType) {
+    case AssessmentItemTypes.MULTIPLE_SELECTION:
+      return answersCopy;
+
+    case AssessmentItemTypes.INPUT_QUESTION:
+      return answersCopy.map(answer => {
+        answer.correct = true;
+        return answer;
+      });
+
+    case AssessmentItemTypes.TRUE_FALSE:
+      return NEW_TRUE_FALSE_ANSWERS;
+
+    case AssessmentItemTypes.SINGLE_SELECTION: {
+      let firstCorrectAnswerIdx = answers.findIndex(answer => answer.correct === true);
+      if (firstCorrectAnswerIdx === -1) {
+        firstCorrectAnswerIdx = 0;
+      }
+
+      const newAnswers = answersCopy.map(answer => {
+        answer.correct = false;
+        return answer;
+      });
+
+      newAnswers[firstCorrectAnswerIdx].correct = true;
+
+      return newAnswers;
+    }
+  }
+};
+
+/**
+ * Sanitize assesment item answers
```

Comment:
Could you expand on what "sanitizing" entails?

Reply:
I've updated docstring. I am not very sure whether "sanitize" is a good word - please feel free to suggest English words that would describe logic more accurately in my PRs any time if you have something in mind :)